packages/zarith/zarith.1.14/opam

```diff
@@ -0,0 +1,58 @@
+opam-version: "2.0"
+maintainer: "Xavier Leroy <xavier.leroy@inria.fr>"
+authors: [
+  "Antoine Miné"
+  "Xavier Leroy"
+  "Pascal Cuoq"
+]
+homepage: "https://github.com/ocaml/Zarith"
+bug-reports: "https://github.com/ocaml/Zarith/issues"
+dev-repo: "git+https://github.com/ocaml/Zarith.git"
+license: "LGPL-2.0-only WITH OCaml-LGPL-linking-exception"
+build: [
+  ["./configure"] {os != "openbsd" & os != "freebsd" & os != "macos"}
+  [
+    "sh"
+    "-exc"
+    "LDFLAGS=\"$LDFLAGS -L/usr/local/lib\" CFLAGS=\"$CFLAGS -I/usr/local/include\" ./configure"
+  ] {os = "openbsd" | os = "freebsd"}
+  [
+    "sh"
+    "-exc"
+    "LDFLAGS=\"$LDFLAGS -L/opt/local/lib -L/usr/local/lib\" CFLAGS=\"$CFLAGS -I/opt/local/include -I/usr/local/include\" ./configure"
+  ] {os = "macos" & os-distribution != "homebrew"}
+  [
+    "sh"
+    "-exc"
+    "LDFLAGS=\"$LDFLAGS -L/opt/local/lib -L/usr/local/lib\" CFLAGS=\"$CFLAGS -I/opt/local/include -I/usr/local/include\" ./configure"
+  ] {os = "macos" & os-distribution = "homebrew" & arch = "x86_64" }
+  [
+    "sh"
+    "-exc"
+    "LDFLAGS=\"$LDFLAGS -L/opt/homebrew/lib\" CFLAGS=\"$CFLAGS -I/opt/homebrew/include\" ./configure"
+  ] {os = "macos" & os-distribution = "homebrew" & arch = "arm64" }
+  [make]
```

Comment:
Now that [`zarith` supports `pkg-config`](https://github.com/ocaml/Zarith/pull/132) I think it may be worth just using that. At least on macos we could simply add `conf-pkg-config` as a dependency and use the same invocation as linux. Maybe this works also on freebsd

Reply:
It is tempting to rely on pkg-config, indeed.  It works very well for developing on Linux and macOS.  But I'm afraid of breaking what seems to be working today...  

At any rate, if we go the pkg-config way, I'd like all the platforms to use it, no special cases if at all possible.  But I realize I don't know how things work for Windows.

Reply:
I don't know about windows, so I would not touch anything there, but according to the tests I made for https://github.com/ocaml/opam-repository/pull/23330 (for all packages with a checkmark I was also testing all the revdeps I could install on my computer) there seems to be no breakage

I would suggest to add a commit to use `["./configure"]` on all systems and then we see from the CI if this works also on freebsd. Ping @kit-ty-kate, do you know if openbsd would work as well or needs a custom call?

Reply:
as far as i know openbsd should work out of the box with pkg-config. The base system ships with pkg-config, its only problem is that it doesn't support some of the arguments but i'm not seeing the use of them in zarith so it should be fine.

Reply:
YOLO.  See latest commit.